OrbitGl/App.cpp

```diff
@@ -552,17 +551,40 @@ ErrorMessageOr<void> OrbitApp::OnLoadPreset(const std::string& filename) {
 void OrbitApp::LoadPreset(const std::shared_ptr<PresetFile>& preset) {
   const int32_t selected_process_id = GetSelectedProcessId();
   const ProcessData* selected_process = data_manager_->GetProcessByPid(selected_process_id);
-  const std::string& process_full_path = preset->preset_info().process_full_path();
-  if (selected_process != nullptr && selected_process->full_path() == process_full_path) {
-    // In case we already have the correct process selected
-    GOrbitApp->LoadModulesFromPreset(GetSelectedProcess(), preset);
-    return;
+  if (selected_process != nullptr) {
+    LoadModulesFromPreset(GetSelectedProcess(), preset);
+  } else {
+    SendErrorToUi("Preset loading failed", "Process is not selected");
   }
-  if (!SelectProcess(Path::GetFileName(process_full_path), preset)) {
-    SendErrorToUi("Preset loading failed",
-                  absl::StrFormat("The process \"%s\" is not running.", process_full_path));
-    return;
+}
+
+PresetLoadState OrbitApp::GetPresetLoadState(
+    const std::shared_ptr<orbit_client_protos::PresetFile>& preset) {
```

Comment:
I think this should be a free (or static) function, taking the selected process as a second argument. Ideally there would also be a test for this free function.

Reply:
But the test could also be a follow-up.

Reply:
I moved the logic to anonymous namespace, ptal